src/main/java/com/github/mizool/jediscache/JedisCache.java

```diff
@@ -0,0 +1,353 @@
+
+import javax.cache.Cache;
+import javax.cache.configuration.CacheEntryListenerConfiguration;
+import javax.cache.configuration.CompleteConfiguration;
+import javax.cache.configuration.Configuration;
+import javax.cache.configuration.MutableConfiguration;
+import javax.cache.integration.CompletionListener;
+import javax.cache.processor.EntryProcessor;
+import javax.cache.processor.EntryProcessorException;
+import javax.cache.processor.EntryProcessorResult;
+
+import lombok.Data;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableMap;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.ScanParams;
+import redis.clients.jedis.ScanResult;
+
+@Slf4j
+@Data
+class JedisCache<K, V> implements Cache<K, V>
+{
+    private final JedisCacheManager cacheManager;
+    private final MutableConfiguration<K, V> configuration;
+    private final JedisPool jedisPool;
+    private final String name;
+    private final JedisCacheKeyConverter<K> keyConverter;
+    private final JedisCacheValueConverter<V> valueConverter;
+    private final byte[] jedisCacheName;
+    private final AtomicBoolean closed;
+
+    public JedisCache(
+        @NonNull JedisCacheManager cacheManager,
+        @NonNull String name,
+        @NonNull Configuration<K, V> configuration,
+        @NonNull JedisPool jedisPool)
+    {
+        this.cacheManager = cacheManager;
+        this.name = name;
+        this.jedisPool = jedisPool;
+
+        //we make a copy of the configuration here so that the provided one
+        //may be changed and/or used independently for other caches. we do this
+        //as we don't know if the provided configuration is mutable
+        if (configuration instanceof CompleteConfiguration)
+        {
+            //support use of CompleteConfiguration
+            this.configuration = new MutableConfiguration<>((CompleteConfiguration<K, V>) configuration);
+        }
+        else
+        {
+            //support use of Basic Configuration
+            MutableConfiguration<K, V> mutableConfiguration = new MutableConfiguration<>();
+            mutableConfiguration.setStoreByValue(configuration.isStoreByValue());
+            mutableConfiguration.setTypes(configuration.getKeyType(), configuration.getValueType());
+            this.configuration = mutableConfiguration;
+        }
+
+        JedisCacheJavaSerializationConverter<K, V>
+            converter
+            = new JedisCacheJavaSerializationConverter<>(configuration.getKeyType(), configuration.getValueType());
+        keyConverter = converter;
+        valueConverter = converter;
+        jedisCacheName = name.getBytes(Charsets.UTF_8);
+
+        closed = new AtomicBoolean(false);
+    }
+
+    @Override
+    public V get(K key)
+    {
+        verifyOpen();
+        try (Jedis jedis = jedisPool.getResource())
```

Comment:
reduce boilerplate code by replacing all these `try` clauses with `try (Jedis jedis = obtainJedis())`, and have that method call `verifyOpen()`